src/main/webapp/availability.js

```diff
@@ -14,4 +14,24 @@
 
 function onAvailabilityLoad() {
   supplyLogoutLink();
+  loadAvailabilityTable();
+}
+
+// Switches a tile from green to white and vice versa when clicked.
+function switchTile(tile) {
+  if(tile.classList.contains('table-success')) {
+    tile.classList.remove('table-success');
+  } else {
+    tile.classList.add('table-success');
+  }
+}
+
+function loadAvailabilityTable() {
+  let date = new Date();
+  let timeZoneOffset = (-1)*date.getTimezoneOffset();
+  console.log(timeZoneOffset);
+  fetch('/availabilityTable.jsp?timeZoneOffset=' + timeZoneOffset).then(response => response.text()).then(table => {
+    console.log(table);
+    document.getElementById('table-container').innerHTML = table;
```

Comment:
In my forumulation of this, I "dependency injected" the document element instead of loading it at the last minute like this. I think that's a better separation of responsibilities. Consider the API

```javascript
loadAvailabilityTable(
    document.getElementById('table-container'),
    getBrowserTimezone());
```

This just "feels" self explanatory, even without looking into the implementation. This also makes writing a test easier.

Reply:
I tried this, returning the table in a separate method that would be set as the HTML of the table-container but when I did it printed undefined. I think it has something to do with the promise chain. I'm open to trying again but would need some guidance on what the method being called would look like (where it would return, basically).

Reply:
> returning the table in a separate method that would be set as the HTML of the table-container but when I did it printed undefined

My suggestion here actually doesn't involve modifying return types. Here is the complete code (% minor bugs)

```javascript

function availabilityTableDiv() {
  return document.getElementById('table-container')
}

function browserTzOffset() {
  let date = new Date();
  return (-1)*date.getTimezoneOffset();
}

function loadAvailabilityTable(tableDiv, tzOffset) {
  fetch('/availabilityTable.jsp?timeZoneOffset=' + tzOffset)
    .then(response => response.text())
    .then(tableContents => {  
      tableDiv.innerHTML = tableContents;
      // You might need an explicit return here, though I doubt it.
      // return
    });
}

function onAvailabilityLoad() {
  supplyLogoutLink();
  loadAvailabilityTable(availabilityTableDiv(), browserTzOffset());
}
```

Reply:
Done